pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -188,47 +197,72 @@ func (r *reconciler) Reconcile(rule *CompletedRule) error {
 	var ofPriority *uint16
 
 	value, exists := r.lastRealizeds.Load(rule.ID)
+	ruleTable := r.getOFRuleTable(rule)
 	if rule.isAntreaNetworkPolicyRule() {
 		// For CNP, only release priorityMutex after rule is installed on OVS. Otherwise,
 		// priority re-assignments for flows that have already been assigned priorities but
 		// not yet installed on OVS will be missed.
-		r.priorityMutex.Lock()
-		defer r.priorityMutex.Unlock()
+		r.priorityMutex[ruleTable].Lock()
+		defer r.priorityMutex[ruleTable].Unlock()
 	}
-	ofPriority, err = r.getOFPriority(rule)
+	ofPriority, err = r.getOFPriority(rule, ruleTable)
 	if err != nil {
 		return err
 	}
 	var ofRuleInstallErr error
 	if !exists {
-		ofRuleInstallErr = r.add(rule, ofPriority)
+		ofRuleInstallErr = r.add(rule, ofPriority, ruleTable)
 	} else {
-		ofRuleInstallErr = r.update(value.(*lastRealized), rule, ofPriority)
+		ofRuleInstallErr = r.update(value.(*lastRealized), rule, ofPriority, ruleTable)
 	}
 	if ofRuleInstallErr != nil && ofPriority != nil {
-		r.priorityAssigner.Release(*ofPriority)
+		r.priorityAssigners[ruleTable].Release(*ofPriority)
 	}
 	return ofRuleInstallErr
 }
 
+// getOFRuleTable retreives the OpenFlow table to install the CompletedRule.
+// The decision is made based on whether the rule is created for a CNP/ANP, and
+// the Tier of that NetworkPolicy.
+func (r *reconciler) getOFRuleTable(rule *CompletedRule) binding.TableIDType {
+	if !rule.isAntreaNetworkPolicyRule() {
+		if rule.Direction == v1beta1.DirectionIn {
+			return openflow.IngressRuleTable
+		} else {
+			return openflow.EgressRuleTable
+		}
+	}
+	var ruleTables []binding.TableIDType
+	if rule.Direction == v1beta1.DirectionIn {
+		ruleTables = openflow.GetCNPIngressTables()
```

Comment:
Questions - what is the plan we add Namespace policy CRD? Will we add separate tables for each tier again?
@abhiraut

Reply:
As far as I'm concerned I think the namespaced policy can share the table with CNP if they are in the same Tier.

Reply:
ok

Reply:
yes ... they should share the same table.. i know it will be hard for users to track Priority across ClusterNetworkPolicy and namespaced NetworkPolicy within a tier.. UI will make it simpler..